pkg/search/postgres/joins.go

```diff
@@ -132,7 +137,17 @@ func getJoinsAndFields(src *walker.Schema, q *v1.Query) ([]innerJoin, map[string
 			field := f
 			lowerCaseName := strings.ToLower(f.Search.FieldName)
 			if unreachedFields.Remove(lowerCaseName) {
-				reachableFields[lowerCaseName] = &field
+				reachableFields[lowerCaseName] = searchFieldMetadata{baseField: &field}
+			}
+			for _, derivedF := range field.DerivedSearchFields {
+				derivedField := derivedF
+				lowerCaseDerivedName := strings.ToLower(derivedField.FieldName)
+				if unreachedFields.Remove(lowerCaseDerivedName) {
+					reachableFields[lowerCaseDerivedName] = searchFieldMetadata{
+						baseField:       &field,
```

Comment:
Why do we add the base field with derived field?

Reply:
Because you need the base field for constructing the query, since that's the column in the DB

Reply:
oh okay. Thanks for clarifying. I thought we already know `DerivedFrom` and `DerivationType`.

Reply:
We do, this is just transferring that information